app/clients/letter/dvla.py

```diff
@@ -80,5 +103,99 @@ def authenticate(self):
+
+        address should be normalised address lines, e.g. ['A. User', 'London', 'SW1 1AA']
+
+        normalised address lines can be returned by calling:
+        `PostalAddress.from_personalisation(notification.personalisation).normalised_lines`
+        """
+        from app.models import INTERNATIONAL_POSTAGE_TYPES
+
+        if postage in INTERNATIONAL_POSTAGE_TYPES:
+            raise NotImplementedError
+
+        current_app.logger.info(f"Sending letter with id {notification_id}")
+
+        try:
+            response = requests.post(
+                f"{current_app.config['DVLA_API_BASE_URL']}/print-request/v1/print/jobs",
+                headers=self._get_auth_headers(),
+                json=self.format_create_print_job_json(
+                    notification_id, address, postage, service_id, organisation_id, pdf_file
+                ),
+            )
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            # Catch errors and raise our own to indicate what action to take.
+            # If the error has details, we add them to the error message.
+            if e.response.status_code == 400:
+                raise DvlaException(response.json()["errors"][0]["detail"])
+            elif e.response.status_code in {401, 403}:
+                raise DvlaUnauthorisedRequestException(
+                    f'{e.response.status_code} {response.json()["errors"][0]["detail"]}'
+                )
+            elif e.response.status_code == 409:
+                raise DvlaDuplicatePrintRequestException(response.json()["errors"][0]["detail"])
+            elif e.response.status_code == 429:
+                raise DvlaThrottlingException(str(e))
+            elif e.response.status_code >= 500:
+                raise DvlaRetryableExcepton(str(e))
+            else:
+                raise DvlaException(str(e))
+        else:
+            return response.json()
+
+    def format_create_print_job_json(
+        self, notification_id, address_lines, postage, service_id, organisation_id, pdf_file
+    ):
+        from app.models import FIRST_CLASS
+
+        recipient_name = address_lines.pop(0)
+
+        json_payload = {
+            "id": notification_id,
+            "standardParams": {
+                "jobType": "NOTIFY",
+                "templateReference": "NOTIFY",
+                "businessIdentifier": notification_id,
+                "recipientName": recipient_name,
+                "address": {"unstructuredAddress": self.build_unstructured_address(address_lines)},
+            },
+            "customParams": [
+                {"key": "pdfContent", "value": base64.b64encode(pdf_file).decode("utf-8")},
+                {"key": "organisationIdentifier", "value": organisation_id},
+                {"key": "serviceIdentifier", "value": service_id},
+            ],
+        }
+
+        if postage == FIRST_CLASS:
+            json_payload["standardParams"]["despatchMethod"] = "FIRST"
```

Comment:
Do we leave this off altogether for 2nd class? I wonder if we should (if we can) be explicit about the despatchMethod in all cases.

Reply:
Just looked at the DVLA docs and I can see why you've done this as their enum for DespatchMethod doesn't include an entry for second. I'm a bit surprised by that...

Reply:
Yes, I did try passing in "SECOND" to see if worked despite not being documented but it gave an error message of "Invalid despatchMethod value"

Reply:
Maybe we can add a comment here just highlighting that detail?